frontend/src/api/doctor.ts

```diff
@@ -0,0 +1,53 @@
+import type { Doctor } from "../components/doctor/DoctorControl";
```

Comment:
I would define the Doctor type here in the api and import it in your components where you need it.